specs/examples/README.md

```diff
@@ -0,0 +1,89 @@
+# Nimble Examples
+
+## Problem Statement
+
+Nimble should have high-level well-maintained examples for our supported frameworks.
+
+This HLD sketches out the organization and requirements of new examples and of the current examples.
+
+## Links To Relevant Work Items and Reference Material
+
+- [#726 Create a "Getting Started" walkthrough for Nimble Angular](https://github.com/ni/nimble/issues/726)
+- [#325 Set up stackblitz or codepen template with Nimble](https://github.com/ni/nimble/issues/325)
+
+## Implementation / Design
+
+### Frameworks to support
+- Vanilla: TypeScript + SCSS
+    - No explicit examples supporting vanilla JavaScript + CSS with local npm builds. Strongly encourage TS + SCSS.
+    - Examples will use [vite](https://vitejs.dev/)
+        - Supports TS + SCSS out of the box
+        - Web Chapter does not have an alternate recommendation yet for Vanilla apps
+        - Vite works on Stackblitz and local npm builds
+            - Parcel [does not work on Stackblitz](https://github.com/parcel-bundler/parcel/issues/7748)
+            - Rollup requires extra config for TS and SCSS
+            - Webpack aligns with the underlying technology of Angular apps but adds lots of complexity to self manage (Angular generally manages for you)
+- CDN: Vanilla JS and CSS
+    - Uses the CDN builds of packages via unpkg.com
+    - Known issue: fonts are not included in the [existing example](https://github.com/ni/nimble/tree/0fe8cc54f7bd444830e3974969519593b532b083/packages/nimble-components#prototyping-in-a-static-webpage)
+        - We should move all the pre-built cdn bundle stuff to a [separate package](https://twitter.com/justinfagnani/status/1521953590931189760?s=20&t=zejlKVWCbEBce8zYFO7Uyg), ie:
+            - `@ni/nimble-bundle`
+                - alternative names:
+                    - `@ni/nimble-components-cdn`
+                    - `@ni/nimbl-components-bundle`
+                    - `@ni/nimble-cdn-bundle`
+                    - `@ni/nimble-cdn`
+        - Alternative:
+            - We could add a build of our CSS + font assets to the existing nimble-components bundle build which will grow the package size and pose the risk of none cdn users accidentally leveraging the contents (as described in the separate package thread linked above).
+- Angular
+- Blazor
+
+### File structure
+  - examples/
+    - `<framework>`/
+        - README.md: Table of contents for the examples
+        - getting-started/
+            - README.md: Should include a Stackblitz link to open the example immediately in addition to other content
+            - package.json
+            - source/
+        - `<other examples>`: TBD
+
```

Comment:
Should we deploy all the built examples to GitHub Pages in addition to being able to run them in Stackblitz? I think that could be useful for doing things like running performance tests on stress test examples and for sending people direct links to demos without the chrome of stackblitz (e.g. for a PO who wants to see what the nimble-table can do but doesn't care what its API is).

If we do this, we should ensure each copy of an example crosslinks to all the other copies: readme <-> hosted example <-> example source code <-> stackblitz <-> published npm package

Reply:
If we do this we might also want to host a Table of Contents site for examples and link to it from nimble.ni.dev.

Reply:
I think I initially lean against that. Our GitHub pages build is very large, ~100MB, due to storybook metadata being included and the bulkiness of individual blazor builds. There is a lot of extra bulk because these are individual app builds so framework overhead is duplicated (vs monorepo where duplicate framework cost is amortized).

> running performance tests on stress test examples 

I don't think if we had automated perf tests we would point to the published GitHub pages example. I think it would be part of the CI pipeline / an npm script you can run at any point by checking out the repo and running perf tests. Testing GitHub pages limits you to just the latest published examples.

Some additional reasons to do it:

- Easier to pull up examples in a mobile environment (i.e. manually open table stress test on phone)
- Nice shareable example URLs

I'll capture it in future work since I think It'll need some research. Maybe stripping out some stuff from the GitHub pages build, etc.